test/jit/test_enum.py

```diff
@@ -135,6 +160,88 @@ def enum_value(x: Color) -> int:
         self.assertEqual(scripted_enum_value(Color.RED), Color.RED.value)
         self.assertEqual(scripted_enum_value(Color.GREEN), Color.GREEN.value)
 
+    def test_enum_as_const(self):
+        global Color
+
+        class Color(Enum):
+            RED = 1
+            GREEN = 2
+
+        @torch.jit.script
+        def enum_const(x: Color) -> bool:
+            if x == Color.RED:
+                return True
+            else:
+                return False
+
+        self.assertEqual(enum_const(Color.RED), True)
+        self.assertEqual(enum_const(Color.GREEN), False)
+
+    def test_non_existent_enum_value(self):
+        global Color
+
+        class Color(Enum):
+            RED = 1
+            GREEN = 2
+
+        def enum_const(x: Color) -> bool:
+            if x == Color.PURPLE:
+                return True
+            else:
+                return False
+
+        with self.assertRaisesRegexWithHighlight(RuntimeError, "has no attribute 'PURPLE'", "Color.PURPLE"):
+            torch.jit.script(enum_const)
+
+    def test_enum_ivalue_type(self):
+        global Color
+
+        class Color(Enum):
+            RED = 1
+            GREEN = 2
+
+        def is_color_enum(x: Any):
+            if isinstance(x, Color):
+                return True
+            else:
+                return False
+
+        with torch._jit_internal._disable_emit_hooks():
+            scripted_is_color_enum = torch.jit.script(is_color_enum)
+
+        self.assertEqual(scripted_is_color_enum(Color.RED), True)
+        self.assertEqual(scripted_is_color_enum(Color.GREEN), True)
+        self.assertEqual(scripted_is_color_enum(1), False)
+
+    def test_closed_over_enum_constant(self):
+        global Color
+
+        class Color(Enum):
+            RED = 1
+            GREEN = 2
+
+        a = Color
+
+        def closed_over_aliased_type():
+            return a.RED.value
+
+        with torch._jit_internal._disable_emit_hooks():
+            scripted = torch.jit.script(closed_over_aliased_type)
+
+        self.assertEqual(scripted(), Color.RED.value)
+
+
+        b = Color.RED
+
```

Comment:
A test to add this time or next time (I think this should work right now) is putting an Enum as a attribute of a module and check that it is retained durign scripting 
`
class Mod(nn.Module): 
     self.a = Color.RED

def forward(self):
     return self.a
```

Reply:
Done.